### PR TITLE
Text File secrets are now stripped of white space and newlines.

### PR DESCRIPTION
**Pull request by jathanism**
# Closes: #2338 
# What's Changed

Self-explanatory.

# TODO

- [ ] Explanation of Change(s)
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/docs/models/extras/secret.md [hunk 1]
```diff
@@ -13,6 +13,9 @@ Each Secret is associated with a secrets provider (not to be confused with a cir
 - *Environment Variable* - for retrieving a secret value defined in an environment variable; Secrets using this provider must specify the `variable` name to retrieve.
 - *Text File* - for retrieving a secret value stored in a text file; Secrets using this provider must specify the absolute `path` of the file to retrieve.
 
+!!! note
+    When using the Text File secrets provider, any trailing white space or newlines will be stripped.
```
Comment (bryanculver) [inline nautobot/docs/models/extras/secret.md:17]:
```suggestion
    When using the Text File secrets provider, any leading and trailing white space or newlines will be stripped.
```

File: nautobot/extras/secrets/providers.py [hunk 2]
```diff
@@ -58,11 +58,14 @@ def clean(self):
 
     @classmethod
     def get_value_for_secret(cls, secret, obj=None, **kwargs):
-        """Retrieve the appropriate text file's contents."""
+        """
+        Retrieve the appropriate text file's contents. The value will be stripped of trailing
+        whitespace and newlines.
```
Comment (bryanculver) [inline nautobot/extras/secrets/providers.py:63]:
```suggestion
        Retrieve the appropriate text file's contents. The value will be stripped of leading
        and trailing whitespace and newlines.
```

Comment (glennmatthews) [inline nautobot/extras/secrets/providers.py:63]:
```suggestion
        """
        Retrieve the appropriate text file's contents.
        
        The value will be stripped of leading and trailing whitespace and newlines.
```

**Review by glennmatthews — APPROVED** [review]

File: nautobot/extras/secrets/providers.py (hunk 2 above)
Comment (bryanculver) [reply nautobot/extras/secrets/providers.py:63]: @glennmatthews's is a better suggestion

**Pull request merged**